driver/bpf/fillers.h

```diff
@@ -6213,8 +6213,8 @@ FILLER(sys_umount_x, true)
 FILLER(sys_umount2_e, true)
 {
 	/* Parameter 1: flags (type: PT_FLAGS32) */
-	u32 flags = (u32)bpf_syscall_get_argument(data, 1);
-	return bpf_push_u32_to_ring(data, flags);
+	int flags = (int)bpf_syscall_get_argument(data, 1);
+	return bpf_push_s32_to_ring(data, flags);
```

Comment:
its correct to handle the flag as an `int` but before pushing it to userspace we need to convert the `umount2` flags to the `PPM_` notation. So something like:
```suggestion
	int flags = (int)bpf_syscall_get_argument(data, 1);
	return bpf_push_u32_to_ring(data, umount2_flags_to_scap(flags));
```
Please note that the converted flags should be positive so `u32`

Reply:
You need to do a similar thing in the kernel module https://github.com/falcosecurity/libs/blob/0347a329fe18c2724d4c6a8ce9a6224105d670e5/driver/ppm_fillers.c#L7268

Reply:
if the flags are s32, the umount2_flags_to_scap should also return s32 (I think)